substratevm/src/com.oracle.svm.configure/src/com/oracle/svm/configure/config/TypeConfiguration.java

```diff
@@ -63,7 +64,44 @@ public ConfigurationType getOrCreateType(String qualifiedForNameString) {
         }
         if (n > 0) { // transform to Java source syntax
             StringBuilder sb = new StringBuilder(s.length() + n);
-            sb.append(s, n + 1, s.length() - 1); // cut off leading '[' and 'L' and trailing ';'
+            if (s.charAt(n) == 'L' && s.charAt(s.length() - 1) == ';') {
+                sb.append(s, n + 1, s.length() - 1); // cut off leading '[' and 'L' and trailing ';'
+            } else {
+                // Primitive Array
+                String primitiveType = s.substring(n, s.length());
+                switch (primitiveType) {
+                    case "I":
+                        s = "int";
+                        break;
+                    case "F":
+                        s = "float";
+                        break;
+                    case "J":
+                        s = "long";
+                        break;
+                    case "D":
+                        s = "double";
+                        break;
+                    case "Z":
+                        s = "boolean";
+                        break;
+                    case "B":
+                        s = "byte";
+                        break;
+                    case "S":
+                        s = "short";
+                        break;
+                    case "C":
+                        s = "char";
+                        break;
+                    case "V":
+                        s = "void";
+                        break;
+                    default:
+                        throw new JSONParserException("Unrecognized primitive type " + primitiveType);
+                }
```

Comment:
Should be possible to replace with `JavaKind.fromPrimitiveOrVoidTypeChar(s.chartAt(n)).getJavaName()` with a check that `n == s.length() - 1`.

Reply:
fixed